app/modules/entities/components/editor/select_author_role.svelte

```diff
@@ -0,0 +1,78 @@
+<script>
+  import { isNonEmptyClaimValue } from '#entities/components/editor/lib/editors_helpers'
+  import { getWorkPreferredAuthorRolesProperties } from '#entities/lib/editor/properties_per_subtype'
+  import preq from '#lib/preq'
+  import { onChange } from '#lib/svelte/svelte'
+  import { I18n, i18n } from '#user/lib/i18n'
+  import { uniq, without } from 'underscore'
+
+  export let entity, property, value
+
+  let rolesProperties
+  function setRolesProperties () {
+    rolesProperties = getWorkPreferredAuthorRolesProperties(entity)
+    if (!rolesProperties.includes(property)) {
+      rolesProperties = [ property, ...rolesProperties ]
+    }
+    if (rolesProperties.length === 1 && rolesProperties[0] === property) {
+      rolesProperties = null
+    }
+  }
+
+  $: onChange(entity, setRolesProperties)
+
+  let currentRoleProperty = property
+
+  async function onRolePropertyChange () {
```

Comment:
not sure `onRolePropertyChange` is the culprit, but when editing [this entity](https://inventaire.io/entity/inv:035a93cc360f4e285e955bc12c282f10/edit), when i select first author "change author role" field to illustrator, then P50 author value (Anne Barber) successfully appear in illustrator prop field, but keeps appearing in author P50 field (and error is thrown if i want to select author P5 again `ERROR: claim property value not found`)

Reply:
Should have been fixed by 228941f0e